File: 03-02-2024_Decimal_Equivalent_of_Binary_Linked_List.cpp
class Solution
{
    public:
        // Should return decimal equivalent modulo 1000000007 of binary linked list 
        long long unsigned int decimalValue(Node *head)
        {
           Node* temp = head;
           long long num = 0;
           while(temp != NULL)
           {
               num = (num * 2 + temp->data) % 1000000007;
               temp = temp->next;
           }
           return num;
        }
};

File: 03-02-2024_Decimal_Equivalent_of_Binary_Linked_List.java
class Solution
{
   long DecimalValue(Node head)
   {
        long num = 0;
        Node temp = head;
        while(temp != null)
        {
            num = (num*2 + temp.data) % 1000000007;
            temp = temp.next;
        }
        return num;
   }
}
